metrics/src/metrics.rs

```diff
@@ -630,7 +663,7 @@ mod test {
         let writer = Arc::new(MockMetricsWriter::new());
         let agent = MetricsAgent::new(writer.clone(), Duration::from_secs(1), 100);
 
-        for i in 0..102 {
+        for i in 0..110 {
```

Comment:
Is the point here just that 110 is > 100? Might be nice to use a constant for the rate and then make this some offset of the max to make it more obvious what we're doing. Could also use the max below in the assert

Reply:
Yes, it is just larger and the difference is more than just 1 or 2, as now both `submit()` and `flush()` upload one "extra" entry that holds the metrics stats.

Good point on adding a constant.  Added.
I think it is more readable this way.